cli/internal/helm/client.go

```diff
@@ -85,10 +85,6 @@ func (c *Client) shouldUpgrade(releaseName, newVersion string, force bool) error
 
 	// This may break for cert-manager or cilium if we decide to upgrade more than one minor version at a time.
 	// Leaving it as is since it is not clear to me what kind of sanity check we could do.
-	if currentVersion == newVersion {
```

Comment:
This was the comment to skip helm upgrade on the same version: 
https://github.com/edgelesssys/constellation/pull/1940#issuecomment-1597132318

Reply:
@malt3 could you elaborate on why you want to skip upgrades to the same version even when the force flag is set?
Seems like a valid use case to me to actually force an upgrade, no matter what

Reply:
The logic we are implementing is actually just converging to a desired state.
If actual == desired, we should not do anything.
The helm upgrade has side effects (like writing a huge folder with backups of custom resources).
The helm client itself likely doesn't do anything if actual == desired, so it has no useful side effects either.

Reply:
This however is assuming that localVersionNumber == clusterVersionNumber is equal to desiredChart == actualChart, which might for some reason not be true.
If I am using a flag that tells me it is "disabling version compatibility checks", I would assume it actually disables all of them, which we would not do in this case (skip the upgrade if versions are equal)

Reply:
Sounds like a reasonable request. I'm fine with always running the helm upgrade if force is set.